data/src/main/java/com/example/data/backendapi/controller/CustomerController.java

```diff
@@ -0,0 +1,22 @@
+package com.example.data.backendapi.controller;
+
+import com.example.data.backendapi.dto.Customer;
+import com.example.data.backendapi.service.CustomerService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+public class CustomerController {
+
+    @Autowired
+    CustomerService customerService;
+
+    @GetMapping("/customers/{id}")
+    Customer one(@PathVariable Long id) {
+
+        return customerService.findById(id)
+                .orElseThrow(() -> new CustomerNotFoundException(id));
```

Comment:
If you like it might be a good practice to add some exertion handler for `CustomerNotFoundException` so the user get sone nice output instead of default spring one.

Also consider the alternative and define your error handler right in the controller. Make sure you clearly understand your options in terms of handling errors.